src/zenml/orchestrators/step_runner.py

```diff
@@ -315,6 +315,11 @@ def _parse_hook_inputs(
             arg_type = annotations.get(arg, None)
             arg_type = resolve_type_annotation(arg_type)
 
+            if isinstance(arg_type, str):
```

Comment:
We don't allow string annotations in any other place for probably the same reason, is there any specific reason we require them for hooks?

This lookup in globals/locals seems pretty unreliable to me, what if a the class that is annotated is not in one of those dicts?